gateway/gateway.go

```diff
@@ -693,6 +704,55 @@ func (gateway *HandleT) getJobDataFromRequest(req *webRequestT) (jobData *jobFro
 	return
 }
 
+func setBatch(body []byte, reqType string) (batch []byte, err error) {
+	body, err = sjson.SetBytes(body, "type", reqType)
+	if err != nil {
+		err = errors.New(response.NotRudderEvent)
+		return
+	}
+	batch, err = sjson.SetRawBytes(BatchEvent, "batch.0", body)
+	if err != nil {
+		err = errors.New(response.NotRudderEvent)
+		return
+	}
+	return
+}
+
+func isNonIdentifiable(anonIDFromReq, userIDFromReq string) bool {
+	if anonIDFromReq == "" && userIDFromReq == "" {
+		return !allowReqsWithoutUserIDAndAnonymousID
+	}
+	return false
+}
+
+func buildUserID(userIDHeader, anonIDFromReq, userIDFromReq string) string {
+	if anonIDFromReq != "" {
+		return userIDHeader + DELIMITER + anonIDFromReq + DELIMITER + userIDFromReq
```

Comment:
Just double checking in case it's a copy/paste issue. You meant the last one to be `userIDFromReq` and not `anonIDFromReq`, right?

Reply:
yep👍